engine/src/regex.cpp

```diff
@@ -364,12 +373,17 @@ regexp *MCR_compile(MCStringRef exp, bool casesensitive)
 	// If the pattern is new, put it in the cache.
 	if (!found)
 	{
-		uint2 i;
-		MCR_free(MCregexcache[PATTERN_CACHE_SIZE - 1]);
-		for (i = PATTERN_CACHE_SIZE - 1 ; i ; i--)
+		// If we don't have a free slot, evict a randomish one.
+		if (t_last_free == -1)
 		{
-			MCregexcache[i] = MCregexcache[i - 1];
+			t_last_free = (uintptr_t(exp) >> 2) % PATTERN_CACHE_SIZE;
```

Comment:
To answer my own question: it's fine to construct integers from pointers, because all pointers can be an integer, but you need a `reinterpret_cast` for going in the other direction.